File: 240409/재귀함수를 이용한 최댓값/maximum-value-with-recursive-function.py
N = int(input())
seq = list(map(int, input().split()))

# # 남의 코드 참고하여 푼 풀이
# def find_max(n):
#     if n == len(seq):
#         return seq[n - 1]

#     return max(seq[n], find_max(n + 1))

# print(find_max(0))

def find_max(n):
    if n == 0:
        return seq[n]

    return max(seq[n], find_max(n - 1))

print(find_max(N - 1))

File: 251227/재귀함수를 이용한 최댓값/maximum-value-with-recursive-function.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> seq;
int find_max(int n);

int main() 
{
    int N;
    cin >> N;

    seq.resize(N);
    for (int i = 0; i < N; i++) 
    {
        cin >> seq[i];
    }

    cout << find_max(N - 1);
    return 0;
}

int find_max(int n) 
{
    if (n == 0) 
    {
        return seq[0];
    }
    return max(seq[n], find_max(n - 1));
}